Fix cross-selling product slider config values

danielm0hr opened the pull request:
The product slider element used in product cross-selling was instantiated with invalid config values. Additionally, the wrong config value and config variable name was used when enabling vertical alignment in the slider.

The same error appeared in other elements which use sliders.

<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?

The product slider element used in the product cross selling twig template used invalid config values. When overwriting the template und configuring the product slider element differently this led to a wrong appearance of the the slider. Additionally, the `has-vertical-alignement` CSS class was added to the slider container based on the wrong config key (`displayMode`), which messed up the whole slider appearance when setting the `boxLayout: image` with `displayMode: standard` for the image. Additionally, as part of enabling vertical alignement, the variable name of the config object was wrong in three different slider elements.

### 2. What does this change do, exactly?

* uses the valid config values for the product slider element
* uses the correct config key (`boxLayout`) to enable vertical alignement
* uses the correct config variable name in all slider elements where this error was present

### 3. Describe each step to reproduce the issue or behaviour.

1. Extend the `product-details/cross-selling/tabs.html.twig` template and configure the product slider with `displayMode: standard` and `boxLayout: image`.

### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [n.a.] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [n.a.] I have written or adjusted the documentation according to my changes
- [n.a.] This change has comments for package types, values, functions, and non-obvious lines of code
- [n.a.] I have read the contribution requirements and fulfil them.

dehare commented:
- Applied PR specsheet in original message
- No demo data available, as this is core functionality.
- Tests (if any) should give unchanged results

dehare commented:
I've updated the description with information about the internals. Please review this PR

danielm0hr committed:
Fix cross-selling product slider config values

The product slider element used in product cross-selling
was instantiated with invalid config values. Additionally,
the wrong config value and config variable name was used
when enabling vertical alignment in the slider.

The same error appeared in other elements which use sliders.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1346) <br/>All committers have signed the CLA.

J-Rahe commented:
Hey @danielm0hr, do i understand that correctly, that your problem is that with your setting the css doesn't go into the 
`      
 // box-image uses cover mode for standard display mode

        .product-image.is-standard 
            height: 100%;
            object-fit: cover;
            font-family: 'object-fit: cover;'; // IE polyfill
        }` 
in the platform/src/Storefront/Resources/app/storefront/src/scss/component/_product-box.scss

philipreinken commented:
Hello, we're closing this pull request due to inactivity. If this change is still important to you, feel free to create a new pull request.

For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline.
